components/collection-table.js

```diff
@@ -0,0 +1,546 @@
+          <strong>Clear URL Columns</strong> restores the shown and hidden
+          columns you have saved to your browser, clearing the URL of hidden
+          columns.
+        </Paragraph>
+      </Instruction>
+    </>
+  )
+}
+
+UrlColumnControls.propTypes = {
+  // Type of collection being displayed
+  collectionType: PropTypes.string.isRequired,
+  // IDs of the hidden columns
+  hiddenColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+  // Called once the user clears the URL columns
+  onClearedUrlHiddenColumns: PropTypes.func.isRequired,
+}
+
+/**
+ * Shows and handles the controls to manage the modal to let the user show and hide individual
+ * columns, and to copy a URL with the currently hidden columns.
+ */
+const BrowserColumnControls = ({
+  columns,
+  hiddenColumns,
+  onChange,
+  onChangeAllHiddenColumns,
+}) => {
+  const className = "mx-1.5 inline h-5 w-5"
+  return (
+    <>
+      <ColumnSelector
+        columns={columns}
+        hiddenColumns={hiddenColumns}
+        onChange={onChange}
+        onChangeAllHiddenColumns={onChangeAllHiddenColumns}
+      />
+      <ColumnUrlCopy hiddenColumns={hiddenColumns} />
+      <Instruction title="Show / Hide Columns and Copy URL Columns">
+        <Paragraph>
+          <strong>Show / Hide Columns</strong> lets you choose which individual
+          columns to show and hide, and lets you show or hide all columns at
+          once, except for the <i>ID</i> column. The
+          <Icon.TableColumnsHidden className={className} />
+          symbol indicates at least one column is hidden. The
+          <Icon.TableColumnsVisible className={className} />
+          symbol indicates all columns are shown.
+        </Paragraph>
+        <Paragraph>
+          <strong>Copy URL Columns</strong> copies a URL with your currently
+          hidden columns attached to it. You can share this URL with others so
+          they can see this collection with the same columns hidden, or you can
+          paste it into another browser.
+        </Paragraph>
+      </Instruction>
+    </>
+  )
+}
+
+BrowserColumnControls.propTypes = {
+  // All available columns
+  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
+  // Column IDs of the hidden columns
+  hiddenColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+  // Called when the user changes individual columns to show or hide
+  onChange: PropTypes.func.isRequired,
+  // Called when the user shows or hides all columns at once
+  onChangeAllHiddenColumns: PropTypes.func.isRequired,
+}
+
+/**
+ * Wraps the table-view column controls, such as the hidden-column selector.
+ */
+const ColumnControls = ({ children }) => {
+  return (
+    <div className="my-1 flex flex-wrap items-center gap-1">{children}</div>
+  )
+}
+
+/**
+ * Renders a single left- or right-pointing scroll indicator that fades out after appearing. Don't
+ * attempt to compose the left-5 and right-5 Tailwind CSS classes or they'll get tree shaken.
+ */
```

Comment:
I like the idea of this scroll indicator but I thought it was clickable when I first saw it and was confused when it wasn't. When I looked at it again I thought it was less confusing. I do think it's important to have some indication of scrollability.

Reply:
I tried to time it so it lasted long enough to see, but too short to click on it.